### PR TITLE
Check no unassigned shards even if the node already left

### PR DESCRIPTION
**Pull request by idegtiarenko**
We are currently checking if there are any unassigned shards on the shutting down node before returning `SingleNodeShutdownMetadata.Status.COMPLETE`. However this check is only triggered if the node is still running. If the node unexpectedly exits the clusters or dies during the migration then we are at risk of loosing shard data of those unassigned shards. 

This change performs the check even if the node is no longer part of the cluster.

Related to: #88635

### CODE REVIEW

File: x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java [hunk 1]
```diff
@@ -426,18 +426,10 @@ public void testNodeShutdownWithUnassignedShards() throws Exception {
...
-        internalCluster().restartNode(nodeA);
-        internalCluster().restartNode(nodeB);
-
-        assertThat(client().admin().cluster().prepareHealth("index").get().getUnassignedShards(), equalTo(1));
-
         putNodeShutdown(nodeAId, SingleNodeShutdownMetadata.Type.REMOVE, null);
 
+        internalCluster().stopNode(nodeA);
+
         assertBusy(() -> assertNodeShutdownStatus(nodeAId, STALLED));
```
Comment (idegtiarenko) [inline x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java:433]: The test was getting correct status, however the node was still part of the cluster.
It is updated to stop the node before it is safe according to the api an still getting the same status.

File: x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportGetShutdownStatusAction.java [hunk 2]
```diff
@@ -192,12 +192,6 @@ static ShutdownShardMigrationStatus shardMigrationStatus(
             );
         }
 
-        // The node is in `DiscoveryNodes`, but not `RoutingNodes` - so there are no shards assigned to it. We're done.
-        if (currentState.getRoutingNodes().node(nodeId) == null) {
-            // We don't know about that node
-            return new ShutdownShardMigrationStatus(SingleNodeShutdownMetadata.Status.COMPLETE, 0);
-        }
```
Comment (idegtiarenko) [inline x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportGetShutdownStatusAction.java:199]: This branch prevents us from checking unassigned shards if the node already left the cluster (causing `currentState.getRoutingNodes().node(nodeId)` to be equal to null)

**Review by ChrisHegarty — APPROVED** [review]
LGTM

File: x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java (hunk 1 above)
Comment (idegtiarenko) [reply x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java:433]: I think it is quiet unusual to have unassigned shards in a stable cluster..
Anyways, I randomized the unit test in [TransportGetShutdownStatusActionTests](https://github.com/elastic/elasticsearch/pull/94722/commits/17c1e46cf623787ec72b8027382d2073d029b92b) to cover both scenarios: shard is unassigned and last node may or may not be in cluster.

**Review by fcofdez — APPROVED** [review]
LGTM

**Pull request merged**